core/http.md

```diff
@@ -0,0 +1,3293 @@
+is set to `false`), then use of the `$details` suffix in a request URL MUST
+be treated the same as if it were absent and any URLs in server response
+messages referencing that Resource (e.g. `self`) MUST NOT include it.
+
+##### Serializing Resource Domain-Specific Documents
+
+When a Resource is serialized as its underlying domain-specific document,
+in other words `$details` is not appended to its URL path, the HTTP body of
+requests and responses MUST be the exact bytes of that document. If the
+document is empty, or there is no document, then the HTTP body MUST be empty
+(zero length).
+
+In this serialization mode, it might be useful for clients to also interact
+with some of the Resource's xRegistry metadata. To support this, some of the
+Resource's xRegistry metadata MAY appear as HTTP headers in  messages.
+
+On responses, unless otherwise stated, all top-level scalar attributes of the
+Resource SHOULD appear as HTTP headers where the header name is the name of the
+attribute prefixed with `xRegistry-`. Note, the optionality of this requirement
+is not to allow for servers to decide whether or not to do so, rather it is to
+allow for [No-Code Servers](#no-code-servers) servers than might not be
+able to control the HTTP response headers.
+
+Top-level map attributes whose values are of scalar types SHOULD also appear as
+HTTP headers (each key having its own HTTP header) and in those cases the
+HTTP header names will be of the form: `xRegistry-<MAPNAME>-<KEYNAME>`.
+Note that map keys MAY contain the `-` character, so any `-` after the 2nd `-`
+is part of the key name. See
+[HTTP Header Values](#http-header-values) for additional information and
+[`labels`](#labels-attribute) for an example of one such attribute.
+
+Certain attributes do not follow this rule if a standard HTTP header name
+is defined for that semantic purpose. See the
+[HTTP-Specific Attribute Processing Rules](#http-specific-attribute-processing-rules)
+section for more information.
+
+Complex top-level attributes (e.g. arrays, objects, non-scalar maps) MUST NOT
+appear as HTTP headers.
+
+On update requests, similar serialization rules apply. However, rather than
+these headers being REQUIRED, the client would only need to include those
+top-level attributes that they would like to change. But, including unchanged
+attributes MAY be done. Any attributes not included in request messages
+MUST be interpreted as a request to leave their values unchanged. Using a
+value of `null` (case-sensitive) indicates a request to delete that attribute.
+
+Any top-level map attributes that appear as HTTP headers MUST be included
+in their entirety and any missing keys MUST be interpreted as a request to
+delete those keys from the map.
+
+Since only some types of attributes can appear as HTTP headers, in order
+to manage the full set of attribute the xRegistry metadata view (via use
+of the `$details` URL suffix) MUST be used instead.
```

Comment:
Unless the resource doesn't support a document, in which case a request without a `$details` suffix should apparently be treated the same way as if `$details` was specified? (This is an example of why I think that rule is unhelpful.)